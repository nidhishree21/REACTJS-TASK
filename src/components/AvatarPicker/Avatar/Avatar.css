img,
.ActiveImgWrapper,
.ListImgWrapper,
.SpinningListImgWrapper {
  border-radius: 50%;
  background-color: transparent;
  color: transparent;
  cursor: pointer;
}

img,
.ActiveImgWrapper:hover,
.ActiveImgWrapper:focus,
.ListImgWrapper,
.SpinningListImgWrapper {
  outline: none;
}

.Img {
  max-width: 100%;
  max-height: 100%;
}

.ActiveImgWrapper {
  max-width: 62px;
  max-height: 62px;
  border: solid 1px transparent;
}

.ActiveImgWrapper:hover,
.ActiveImgWrapper:focus {
  border: solid 1px rgb(155, 160, 163);
}

.ActiveImgWrapper::-moz-focus-inner,
.ListImgWrapper::-moz-focus-inner,
.SpinningListImgWrapper::moz-focus-inner {
  border: 0;
}

.ListImgWrapper,
.SpinningListImgWrapper {
  position: relative;
  max-width: 49px;
  max-height: 49px;
  border: solid 3px transparent;
}

.SpinningListImgWrapper:before {
  position: absolute;
  top: -3px;
  left: -3px;
  width: 100%;
  height: 100%;
  content: "";
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  border-color: white rgb(122, 161, 178) rgb(122, 161, 178) rgb(122, 161, 178);
  animation: spin 1.5s linear infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.ListImgWrapper:before {
  content: "";
  position: absolute;
  top: -3px;
  left: -3px;
  width: 100%;
  height: 100%;
  border: solid 3px rgba(155, 160, 163, 0);
  border-radius: 50%;
}

.ListImgWrapper:hover:before,
.ListImgWrapper:focus:before {
  box-sizing: border-box;
  background-color: rgba(122, 161, 178, 0.2);
  border: solid 3px rgba(155, 160, 163, 1);
  width: 49px;
  height: 49px;
}

.HighlightedListImgWrapper {
  border: solid 3px rgb(122, 161, 178) !important;
}
