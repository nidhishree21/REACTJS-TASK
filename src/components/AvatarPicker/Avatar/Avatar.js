import React from "react";
import PropTypes from "prop-types";
import styles from "./Avatar.css";

export const getAvatarStyles = (props, styles) => {
  const { isActive, highlight, showSpinner } = props;
  if (isActive) {
    return styles.ActiveImgWrapper;
  }

  if (!isActive && showSpinner) {
    return styles.SpinningListImgWrapper;
  }

  if (!isActive) {
    return `${styles.ListImgWrapper} ${
      highlight ? styles.HighlightedListImgWrapper : null
    }`;
  }
};

const Avatar = props => {
  const { avatar, onClickAvatar } = props;
  return (
    <div
      tabIndex="0"
      onClick={onClickAvatar}
      onKeyUp={onClickAvatar}
      id={avatar.id}
      className={getAvatarStyles(props, styles)}
    >
      <img className={styles.Img} alt={avatar.label} src={avatar.src} />
    </div>
  );
};

Avatar.propTypes = {
  avatar: PropTypes.shape({
    src: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired
  }),
  onClickAvatar: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  highlight: PropTypes.bool.isRequired,
  showSpinner: PropTypes.bool.isRequired
};

export default Avatar;
