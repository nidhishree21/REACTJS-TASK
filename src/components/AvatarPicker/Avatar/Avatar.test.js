import React from "react";
import { mount } from "enzyme";
import Avatar, { getAvatarStyles } from "./Avatar";

const testProps = {
  avatar: {
    src: "/static/media/avatar1.7467b519.png",
    label: "Avatar 1",
    id: 1
  },
  onClickAvatar: jest.fn(() => {}),
  isActive: true,
  highlight: false,
  showSpinner: false
};

const mountAvatar = (props = {}) => {
  const p = { ...testProps, ...props };
  return mount(<Avatar {...p} />);
};

describe("getAvatarStyles", () => {
  const testStyles = {
    ActiveImgWrapper: "ActiveImgWrapper",
    ListImgWrapper: "ListImgWrapper",
    HighlightedListImgWrapper: "HighlightedListImgWrapper"
  };
  const f = (p, s) => getAvatarStyles(p, s);

  it("returns correct styles for active avatar", () => {
    const s = testStyles.ActiveImgWrapper;
    const p = { ...testProps };
    expect(f(p, testStyles) === s).toBe(true);
  });

  it("returns correct styles for list avatar", () => {
    const s = `${testStyles.ListImgWrapper} ${null}`;
    const p = { ...testProps };
    p.isActive = false;
    expect(f(p, testStyles) === s).toBe(true);
  });

  it("returns correct styles for list avatar", () => {
    const s = `${testStyles.ListImgWrapper} ${
      testStyles.HighlightedListImgWrapper
    }`;
    const p = { ...testProps };
    p.isActive = false;
    p.highlight = true;
    expect(f(p, testStyles) === s).toBe(true);
  });

  it("returns correct styles for list avatar - spinning", () => {
    const s = testStyles.SpinningListImgWrapper;
    const p = { ...testProps };
    p.isActive = false;
    p.showSpinner = true;
    expect(f(p, testStyles) === s).toBe(true);
  });
});

describe("Avatar", () => {
  const wrapper = mountAvatar();

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onClickAvatar function on clicking the active Avatar", () => {
    const avatarToClick = wrapper.find("div");
    avatarToClick.simulate("click");
    expect(testProps.onClickAvatar).toHaveBeenCalledTimes(1);
  });
});
