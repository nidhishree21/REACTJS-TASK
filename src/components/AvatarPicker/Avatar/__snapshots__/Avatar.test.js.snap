// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<Avatar
  avatar={
    Object {
      "id": 1,
      "label": "Avatar 1",
      "src": "/static/media/avatar1.7467b519.png",
    }
  }
  highlight={false}
  isActive={true}
  onClickAvatar={[Function]}
  showSpinner={false}
>
  <div
    className="ActiveImgWrapper"
    id={1}
    onClick={[Function]}
    onKeyUp={[Function]}
    tabIndex="0"
  >
    <img
      alt="Avatar 1"
      className="Img"
      src="/static/media/avatar1.7467b519.png"
    />
  </div>
</Avatar>
`;
